### PR TITLE
Unit Test for handler package

### PR DESCRIPTION
**Pull request by ternakkode**
### SECTION 1: RESOURCE SERVER (http://localhost:5001/)
- [ ] http://localhost:5001/ home page loads
- [ ] "LoginAsUser" button takes you to the user login page
- [ ] Log in using "tester" and "12341234" should succeed
- [ ] Updating the display name should work
- [ ] Log out and Register a new user and try to log in with that new user
- [ ] "LoginAsClient" button takes you to the client login page
- [ ] Log in using "layer8" and "12341234" should work
- [ ] You will be able to see your UID and Secret in profile
- [ ] Log out and Register a new client and log in using that should work

### SECTION 2: WE'VE GOT POEMS (http://localhost:5173/)
- [ ] Log in with 'tester' / '1234'
- [ ] Log in anonymously with Layer8
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3
- [ ] Clicking "Logout" takes you to the login screen
- [ ] Clicking "Register" takes you to the registration page
- [ ] Registering with a username, password, and profile image is successful
- [ ] Logging in with the new username / password succeeds
- [ ] Logging in with Layer8 opens the pop-up
- [ ] Logging in with the newly registered credentials from Section 1 succeeds
- [ ] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3
- [ ] Clicking "Logout" takes you to the login page
- [ ] Log in with 'tester' / '1234', and then with the credentials from Section 1 succeeds
- [ ] This time, DO NOT share "display name" or "country"
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3

### SECTION 3: IMSHARER (http://localhost:5174/)
- [ ] Main page loads
- [ ] Upload of image works
- [ ] Reload leads to instant reload (demonstrating proper caching)
- [ ] Clicking the newly loaded image shows it in a lightbox

### CODE REVIEW

File: server/handlers/authentication_test.go
Comment (stravid87) [reply server/handlers/authentication_test.go:219]: I see that  `parseLoginWithErr` is a generic error handler. Could it be made dynamic? 
```
func (a *authenticationHandlerImpl) parseLoginWithErr(w http.ResponseWriter, r *http.Request, err error, status int) {
    w.WriteHeader(status)
    
    [...]
}
```

File: server/internals/service/service.go
Comment (stravid87) [inline server/internals/service/service.go:22]: I like it. In general, coding to an interface is a good way to avoid many future headaches.

**Review by stravid87 — CHANGES_REQUESTED** [review]
Great work. I really learned a lot from this PR. I especially appreciated learning about the mockgen framework and seeing how it could be used. In a general sense, I starting to be of the opinion that testing, in some ways, demands a better understanding of the program then does writing code where the developer can focus on the linear sequence of events pertaining ONLY to their code. By contrast, testing requires a thorough understanding of the entire project and where your code fits in. It's more challenging in some ways and, I find, often requires more mental space to keep in mind everything happening and what you *expect* to happen. Great work. 

Of 13 comments, I think ~half are just comments and the other are requested changes. Because some of the changes affect the code base, I will be waiting for Veronika to merge her PR first +/- Huzaifa depending on how far a long he is. Otherwise, they will need to retest their entire branches after reverse merging your changes here.

File: server/handlers/authentication.go
Comment (ternakkode) [reply server/handlers/authentication.go:91]: It is the existing behavior to return 200, but we can change to 401

File: server/handlers/authentication_test.go
Comment (ternakkode) [reply server/handlers/authentication_test.go:219]: We can have that, but since the existing code didn't require a change in the status code, I decided not to include it.

File: server/handlers/authentication_test.go
Comment (ternakkode) [reply server/handlers/authentication_test.go:202]: Nil and zero value is usually streated as same value in golang, but let me change it to return `models.LoginUserResponseOutput{}` so the mock have the same behavior.

File: server/handlers/authentication.go
Comment (ternakkode) [reply server/handlers/authentication.go:41]: Next is not a pointer, it can't have a nil value. However, if the value is a special character, it needs to be tested via an integration test because it relates to browser behavior handling redirection to special character URL

**Pull request closed without merging**
